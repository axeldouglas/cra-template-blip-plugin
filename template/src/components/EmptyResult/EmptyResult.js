import React from 'react';
import PropTypes from 'prop-types';
import { ReactComponent as EmptyImg } from '../../assets/images/svg/empty.svg';

const EmptyResult = ({ title, description, children }) => {
    const renderTitle = () => (
        <h5 className="f4 bp-c-medium-elephant mt4 tc">{title}</h5>
    );
    const renderDescription = () => (
        <p className="f5 bp-c-medium-cloud mt2 mw6 tc">
            {description || children}
        </p>
    );

    return (
        <div className="flex flex-column items-center justify-center">
            <EmptyImg />
            {!!title && renderTitle()}
            {(!!description || !!children) && renderDescription()}
        </div>
    );
};

EmptyResult.propTypes = {
    title: PropTypes.string,
    description: PropTypes.string,
    children: PropTypes.any
};

export default EmptyResult;
