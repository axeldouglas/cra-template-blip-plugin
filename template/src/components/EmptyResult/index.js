import EmptyResult from './EmptyResult';

export default EmptyResult;
