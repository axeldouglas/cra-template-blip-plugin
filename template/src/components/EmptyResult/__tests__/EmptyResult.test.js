import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, screen } from '@testing-library/react';

import EmptyResult from '../EmptyResult';

describe('Empty result page', () => {
    it('should render correctly', () => {
        render(
            <EmptyResult
                title="Nenhum plugin encontrado"
                description="Favor buscar novamente"
            />
        );

        expect(
            screen.getByRole('heading', {
                name: /nenhum plugin encontrado/i
            })
        ).toBeInTheDocument();
        expect(screen.getByText(/favor buscar novamente/i)).toBeInTheDocument();
    });

    it('should matches with snapshot', () => {
        const { container } = render(
            <EmptyResult
                title="Nenhum plugin encontrado"
                description="Favor buscar novamente"
            />
        );
        expect(container.firstChild).toMatchSnapshot();
    });
});
