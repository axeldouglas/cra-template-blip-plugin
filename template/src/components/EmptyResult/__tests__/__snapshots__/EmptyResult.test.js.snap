// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty result page should matches with snapshot 1`] = `
<div
  class="flex flex-column items-center justify-center"
>
  <svg>
    empty.svg
  </svg>
  <h5
    class="f4 bp-c-medium-elephant mt4 tc"
  >
    Nenhum plugin encontrado
  </h5>
  <p
    class="f5 bp-c-medium-cloud mt2 mw6 tc"
  >
    Favor buscar novamente
  </p>
</div>
`;
